src/test/java/org/jetbrains/plugins/ideavim/ex/implementation/services/VimVariableServiceTest.kt

```diff
@@ -113,4 +115,35 @@ class VimVariableServiceTest : VimTestCase() {
     typeText("n")
     assertState("1\n2\n1\n2\n1\n2\n1\n2\n${c}1\n")
   }
+
+  @Test
+  fun `test v register`() {
+    configureByText("abcd")
+    // Check default register
+    typeText(commandToKeys("echo v:register"))
+    assertExOutput("\"")
+    // Check default register in expression
+    enterCommand(
+      """
+      nnoremap <expr> X ':echo v:register<CR>'
+      """.trimIndent(),
+    )
+    typeText("X")
+    assertExOutput("\"")
+    // Check named register
+    typeText("\"w")
+    typeText(commandToKeys("echo v:register"))
+    assertExOutput("w")
+    // Check named register in expression
+    // Here the key is that the register is defined as part of the expression
+    // itself, i.e. it hasn't been committed yet as the current register
+    enterCommand(
+      """
+      vnoremap <expr> y '"' . v:register . 'y'
+      """.trimIndent(),
+    )
+    typeText("vl\"zy")
+    val register = injector.registerGroup.getRegisters().filter { reg -> reg.name == 'z' }.first()
+    assertEquals("ab", register.text)
+  }
```

Comment:
I think it would be a good idea to extract this into a new class (perhaps in a new `variables` folder), and split down into separate functions. Hopefully, we'll implement more variables, and it'll save this test class getting overcrowded. It's also good to have tests that test a single thing.